src/main/java/software/amazon/awssdk/crt/io/ClientBootstrap.java

```diff
@@ -14,15 +14,36 @@
  */
 package software.amazon.awssdk.crt.io;
 
+import java.util.concurrent.CompletableFuture;
 import software.amazon.awssdk.crt.CrtResource;
 import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.Log;
 
 /**
  * This class wraps the aws_client_bootstrap from aws-c-io to provide
  * a client context for all protocol stacks in the AWS Common Runtime.
  */
 public final class ClientBootstrap extends CrtResource {
 
+    private final CompletableFuture<Void> shutdownComplete = new CompletableFuture<>();
+
+    /**
+     * Creates a new ClientBootstrap. Most applications will only ever need one instance of this.
+     * @param numThreads The number of Threads to use in the EventLoop
+     * @throws CrtRuntimeException If the system is unable to allocate space for a native client bootstrap object
+     */
+    public ClientBootstrap(int numThreads) throws CrtRuntimeException {
+        try(EventLoopGroup elg = new EventLoopGroup(numThreads);
+            HostResolver hr = new HostResolver(elg)) {
```

Comment:
But I removed all the automagic creation of resources from Python just the other week?
On the suspicion that it would bite users in the butt if we're automatically creating big heavy resources on their behalf without them realizing it?

Reply:
Looks like I removed all the usages of this overload but forgot to delete it afterwards

Reply:
But it looks like NEW code in this PR. Maybe a bad merge brought it back to life

Reply:
It did, I'll remove it.

Reply:
it was a confusing merge.

Reply:
I have no idea what you're talking about.